lib/runner.js

```diff
@@ -0,0 +1,83 @@
+const p = require('..')
+const cf = require('clownface')
+const rdf = require('rdf-ext')
+const namespace = require('@rdfjs/namespace')
+const TextLogStream = require('../lib/TextLogStream')
+
+const ns = {
+  p: namespace('https://pipeline.described.at/'),
+  rdf: namespace('http://www.w3.org/1999/02/22-rdf-syntax-ns#')
+}
+
+function guessPipeline (dataset) {
+  const graph = cf(dataset)
+
+  const pipelines = graph.has(ns.rdf('type'), [ ns.p('Pipeline'), ns.p('ObjectPipeline') ])
+
+  if (pipelines.values.length === 0) {
+    throw new Error('no pipeline found in the dataset')
+  }
+
+  const rootPipelines = pipelines.values.reduce((arr, id) => {
+    const node = dataset.match(null, null, rdf.namedNode(id), null)
+
+    if (node.length === 0) {
+      arr.push(id)
+    }
+
+    return arr
+  }, [])
+
+  if (rootPipelines.length > 1) {
+    throw new Error('multiple root pipeline found. please specify the one to run using --pipeline option')
+  }
+
+  return rootPipelines[0]
+}
```

Comment:
I think guessing the right pipeline should stay in the cli code (or some util function) and in the API the `pipeline` argument should be required. That removes some `magic` from the logic and makes it more understandable and easier to document.